backend/packages/Upgrade/src/api/controllers/ExperimentClientController.v6.ts

```diff
@@ -0,0 +1,894 @@
+   *           schema:
+   *             type: object
+   *             properties:
+   *               group:
+   *                 type: object
+   *                 properties:
+   *                   schoolId:
+   *                      type: array
+   *                      items:
+   *                        type: string
+   *                        example: school1
+   *                   classId:
+   *                      type: array
+   *                      items:
+   *                        type: string
+   *                        example: class1
+   *                   instructorId:
+   *                      type: array
+   *                      items:
+   *                        type: string
+   *                        example: instructor1
+   *               workingGroup:
+   *                 type: object
+   *                 properties:
+   *                   schoolId:
+   *                      type: string
+   *                      example: school1
+   *                   classId:
+   *                      type: string
+   *                      example: class1
+   *                   instructorId:
+   *                      type: string
+   *                      example: instructor1
+   *           description: ExperimentUser
+   *       tags:
+   *         - Client Side SDK
+   *       produces:
+   *         - application/json
+   *       responses:
+   *          '200':
+   *            description: Set Group Membership
+   *            schema:
+   *              $ref: '#/definitions/v6initResponse'
+   *          '400':
+   *            description: BadRequestError - InvalidParameterValue
+   *          '401':
+   *            description: AuthorizationRequiredError
+   *          '404':
+   *            description: Experiment User not defined
+   *          '500':
+   *            description: Internal Server Error
+   */
+  @Post('init')
+  public async init(
+    @Req()
+    request: AppRequest,
+    @Body({ validate: true })
+    experimentUser: ExperimentUserValidatorv6
+  ): Promise<Pick<ExperimentUser, 'id' | 'group' | 'workingGroup'>> {
+    request.logger.info({ message: 'Starting the init call for user' });
+    // getOriginalUserDoc call for alias
+    const experimentUserDoc = request.userDoc;
+    // if reinit call is made with any of the below fields not included in the call,
+    // then we will fetch the stored values of the field and return them in the response
+    // for consistent init response with 3 fields ['userId', 'group', 'workingGroup']
+    const { id = request.get('User-Id'), group, workingGroup } = { ...experimentUserDoc, ...experimentUser };
+    if (experimentUserDoc) {
+      // append userDoc in logger
+      request.logger.child({ userDoc: experimentUserDoc });
+      request.logger.info({ message: 'Got the original user doc' });
+    }
+
+    const upsertResult = await this.experimentUserService.upsertOnChange(
+      experimentUserDoc,
+      experimentUser,
```

Comment:
The id should be added in the experimentUser

Reply:
Yes, that's done in upsertOnChange() as here the validator for experimentUser doesn't have id property. That's why I have to pass the id as an additional optional parameter to this upsertOnChange() function.

Reply:
You can pass the data as `{id, ...experimentUserDoc}`. The type on the `upsertChange` does expect the ID in the experimentUserDoc.